### DISCUSSION TITLE
Scrolling redraw

### DISCUSSION THREAD

**Discussion by PhillipOConnor** [fltk-rs/fltk-rs#1294] [Q&A] [answered]
I have a resizeable browser widget with some lines long enough to bring up a horizontal scrollbar when the widget is narrowed slightly. If I then use the scrollbar the browser contents is redrawn over the old image resulting in a doubled or smeared image. If I resize the browser slightly the image is redrawn correctly as the size change causes a redraw.

I assume I could fix this with a redraw but I cannot find a hook to know the scrollbar was used. Is there a way to scroll without messing up the display?

I believe I am using current crates under windows.
`fltk = "1.3.16"`
`fltk-theme = "0.4.2"`

**Comment by MoAlyousef** [comment, marked as answer]
It might be a FrameType issue. Do you change the frame type of the browser?
Do you have a minimal repro?

**Reply by PhillipOConnor** [reply]
Thanks, while creating a minimal example I removed the frame and the example worked.

The example also helped me find why I could not resize smaller than some minimum, the initial window position has a side effect of setting how small a window can be resized. A window created by `new(100,100,400,300,"Test")` will resize smaller than one created by `new(200,200,400,300,"Test")`.

Is there a list of FLTK oddities? The C++ library seems to have some traps for new players.

**Reply by MoAlyousef** [reply]
Some oddities can be found here:
https://www.fltk.org/doc-1.4/osissues.html

Generally I would say these occur because FLTK abstracts platform-specific behavior, which is sometimes difficult to unify. Windowing and event-handling discrepancies can happen. 

Regarding the frametype issue, it’s because frametypes containing Frame (as opposed to Box) in them are not considered opaque (similar to NoBox), so damage (scrolling, pushing if it’s a button) can produce an artifact:
https://github.com/fltk/fltk/issues/263
